services/apps/src/apps/service/apps/jobs/sharing.clj

```diff
@@ -173,16 +173,28 @@
    (catch ce/clj-http-error? {:keys [body]}
      (str "unable to unshare result folder: " (:error_code (service/parse-json body))))))
 
-;; The apps client isn't used at this time, but it will be once we extend analysis sharing
-;; to HPC apps.
+(defn- unshare-input-file
+  [sharer sharee path]
+  (try+
+   (data-info/unshare-path sharer path sharee)
```

Comment:
Yeah, that's a good question. It would cause the unsharing operation to fail in most cases, which is not really correct. Unfortunately, there's no way to determine whether or not the sharee had access to the file before the analysis was originally shared. Possibly a better option would be only to attempt to unshare the file if the sharer is the owner of the file. During a sharing operation, this is handled by the file sharing service, which doesn't attempt to share the path if the sharee already has access to it.

I think I'm going to leave it as it is for the time being and create a Jira issue to modify the behavior of the unsharing endpoint in the next release.